TACC-Cloud/geoapi
# Task/WG-212: improve projects endpoint

## Overview: ##

The Hazmapper client does the following when creating a project:
- A) request to geoapi to create a map project
- B) request to Tapis to write a file and delete file (i.e. .hazmapper)
- C) request to Tapis (which gets passed onto DesignSafe) to update metadata for the project
- D) creates 2 default tile layers

This PR:

- [x] Refactors the payload for the create project route
- [x] Fixes issues when determining admin and creator status after project is created
- [x] Moves (B) to backend
- [x] Moves (C) to backend
- [x] Moves (D) to backend

and Hazmapper is updated in https://github.com/TACC-Cloud/hazmapper/pull/199

## Related Jira tickets: ##

* [WG-212](https://tacc-main.atlassian.net/browse/WG-212)

## Testing Steps: ##
1. Run this locally
2. Test with https://github.com/TACC-Cloud/hazmapper/pull/199 (Angular code so `cd angular`, use the correct node version and `npm run start`).
3. Create new map projects
4. Ensure that two base layers are created by default
5. Ensure that a .hazmpapper file is created
6. if connected to a DS project, ensure that the metadata for that DS project has been updated.



## Notes: ##
